src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -45,20 +47,46 @@ const postCompany = async (request, h) => {
   const form = forms.handleRequest(selectCompanyForm(request, company), request, schema);
   if (form.isValid) {
     const redirectPath = helpers.processCompanyFormData(request, regionId, companyId, form);
-    return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/${redirectPath}`);
+    return h.redirect(redirectPath);
   }
   const { viewData: { redirectPath } } = dataService.sessionManager(request, regionId, companyId);
   return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId), { redirectPath });
 };
 
+// Handles the redirection to the address entry flow
+const getCreateAddress = async (request, h) => {
+  const { regionId, companyId } = request.params;
+  const queryTail = queryString.stringify({
+    redirectPath: `/invoice-accounts/create/${regionId}/${companyId}/address-entered`,
+    back: `/invoice-accounts/create/${regionId}/${companyId}/check-details`
```

Comment:
I think back link path should be passed in here so that if it comes from the check details it goes back there or if it comes from the select an address form in the flow it should go back there